src/Enterspeed.Source.UmbracoCms.V8/EventHandlers/EnterspeedMediaItemSavedEventHandler.cs

```diff
@@ -0,0 +1,89 @@
+using System.Collections.Generic;
+using System.Linq;
+using Enterspeed.Source.UmbracoCms.V8.Data.Models;
+using Enterspeed.Source.UmbracoCms.V8.Data.Repositories;
+using Enterspeed.Source.UmbracoCms.V8.Factories;
+using Enterspeed.Source.UmbracoCms.V8.Services;
+using Umbraco.Core.Composing;
+using Umbraco.Core.Events;
+using Umbraco.Core.Models;
+using Umbraco.Core.Scoping;
+using Umbraco.Core.Services;
+using Umbraco.Core.Services.Implement;
+using Umbraco.Web;
+using Umbraco.Web.Security;
+
+namespace Enterspeed.Source.UmbracoCms.V8.EventHandlers
+{
+    public class EnterspeedMediaItemSavedEventHandler : BaseEnterspeedEventHandler, IComponent
+    {
+        private readonly IEnterspeedJobFactory _enterspeedJobFactory;
+        private readonly IMediaService _mediaService;
+
+        public EnterspeedMediaItemSavedEventHandler(
+            IUmbracoContextFactory umbracoContextFactory,
+            IEnterspeedJobRepository enterspeedJobRepository,
+            IEnterspeedJobsHandlingService jobsHandlingService,
+            IEnterspeedConfigurationService configurationService,
+            IScopeProvider scopeProvider,
+            IEnterspeedJobFactory enterspeedJobFactory,
+            IMediaService mediaService
+        )
+            : base(
+                umbracoContextFactory,
+                enterspeedJobRepository,
+                jobsHandlingService,
+                configurationService,
+                scopeProvider)
+        {
+            _enterspeedJobFactory = enterspeedJobFactory;
+            _mediaService = mediaService;
+        }
+
+        public void Initialize()
+        {
+            MediaService.Saved += MediaService_Saved;
+        }
+
+        private void MediaService_Saved(IMediaService sender, SaveEventArgs<IMedia> e)
+        {
+            var isPublishConfigured = _configurationService.IsPublishConfigured();
+
+            if (!isPublishConfigured)
+            {
+                return;
+            }
+
+            var entities = e.SavedEntities.ToList();
+            var jobs = new List<EnterspeedJob>();
+
+            foreach (var mediaItem in entities)
+            {
+                if (mediaItem.ContentType.Alias.Equals("Folder"))
+                {
+                    var mediaItems = _mediaService.GetPagedDescendants(mediaItem.Id, 0, 99999, out var totalRecords).ToList();
```

Comment:
Is there a specific reason for using `9999`? Can we make it less magical number as const?